yt/frontends/athena/data_structures.py

```diff
@@ -632,12 +633,18 @@ def _parse_parameter_file(self):
 
     @classmethod
     def _is_valid(cls, filename, *args, **kwargs):
-        try:
-            if "vtk" in filename:
-                return True
-        except Exception:
-            pass
-        return False
+        if not filename.endswith(".vtk"):
+            return False
+
+        with open(filename, "rb") as fh:
+            if not re.match(b"# vtk DataFile Version \\d\\.\\d\n", fh.readline(256)):
+                return False
+            if not re.match(
+                b"(CONSERVED|PRIMITIVE) vars at time= .*, level= \\d, domain= \\d\n",
```

Comment:
I need to dig into the code and relevant datasets more, but I recall that there are some Athena datasets which are just one variable only--so they may not have `CONSERVED` or `PRIMITIVE` in the header. It may take me a day or two to figure out if this is the case.

Reply:
Good to know ! it may be useful to have such a dataset in the sample collection. I wrote using only the ones we already have as validation.

Reply:
I checked, you remembered correctly:
https://princetonuniversity.github.io/Athena-Cversion/AthenaDocsUGbtk

From the [source code](https://www.astro.princeton.edu/~jstone/Athena/doxygen/html.with_source/) (see line 129), it looks like in the case of single-variables, "CONSERVED vars" is replaced by "Really cool Athena data". I could reproduce this joke and hardcode it in yt too, but it seems safer to assume it could change any time in the future (no matter how unlikely), so I'm going to loosen the strictness of this regexp.

Reply:
Fine by me.